libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/Common/phyHandling.c

```diff
@@ -0,0 +1,687 @@
+	{
+	BaseType_t xPhyAddress = pxPhyObject->ucPhyIndexes[ xPhyIndex ];
+	uint32_t ulPhyID = pxPhyObject->ulPhyIDs[ xPhyIndex ];
+
+		/* Write advertise register. */
+		pxPhyObject->fnPhyWrite( xPhyAddress, phyREG_04_ADVERTISE, ulAdvertise );
+
+		/*
+				AN_EN        AN1         AN0       Forced Mode
+				  0           0           0        10BASE-T, Half-Duplex
+				  0           0           1        10BASE-T, Full-Duplex
+				  0           1           0        100BASE-TX, Half-Duplex
+				  0           1           1        100BASE-TX, Full-Duplex
+				AN_EN        AN1         AN0       Advertised Mode
+				  1           0           0        10BASE-T, Half/Full-Duplex
+				  1           0           1        100BASE-TX, Half/Full-Duplex
+				  1           1           0        10BASE-T Half-Duplex
+												   100BASE-TX, Half-Duplex
+				  1           1           1        10BASE-T, Half/Full-Duplex
+												   100BASE-TX, Half/Full-Duplex
+		*/
+
+		/* Read Control register. */
+		pxPhyObject->fnPhyRead( xPhyAddress, phyREG_00_BMCR, &ulConfig );
+
+		ulConfig &= ~( phyBMCR_SPEED_100 | phyBMCR_FULL_DUPLEX );
+
+		ulConfig |= phyBMCR_AN_ENABLE;
+
+		if( ( pxPhyProperties->ucSpeed == ( uint8_t )PHY_SPEED_100 ) || ( pxPhyProperties->ucSpeed == ( uint8_t )PHY_SPEED_AUTO ) )
+		{
+			ulConfig |= phyBMCR_SPEED_100;
+		}
+		else if( pxPhyProperties->ucSpeed == ( uint8_t )PHY_SPEED_10 )
+		{
+			ulConfig &= ~phyBMCR_SPEED_100;
+		}
+
+		if( ( pxPhyProperties->ucDuplex == ( uint8_t )PHY_DUPLEX_FULL ) || ( pxPhyProperties->ucDuplex == ( uint8_t )PHY_DUPLEX_AUTO ) )
+		{
+			ulConfig |= phyBMCR_FULL_DUPLEX;
+		}
+		else if( pxPhyProperties->ucDuplex == ( uint8_t )PHY_DUPLEX_HALF )
+		{
+			ulConfig &= ~phyBMCR_FULL_DUPLEX;
+		}
+
+		if( xHas_19_PHYCR( ulPhyID ) )
+		{
+		uint32_t ulPhyControl;
+			/* Read PHY Control register. */
+			pxPhyObject->fnPhyRead( xPhyAddress, phyREG_19_PHYCR, &ulPhyControl );
+
+			/* Clear bits which might get set: */
+			ulPhyControl &= ~( PHYCR_MDIX_EN|PHYCR_MDIX_FORCE );
+
+			if( pxPhyProperties->ucMDI_X == PHY_MDIX_AUTO )
+			{
+				ulPhyControl |= PHYCR_MDIX_EN;
+			}
+			else if( pxPhyProperties->ucMDI_X == PHY_MDIX_CROSSED )
+			{
+				/* Force direct link = Use crossed RJ45 cable. */
+				ulPhyControl &= ~PHYCR_MDIX_FORCE;
+			}
+			else
+			{
+				/* Force crossed link = Use direct RJ45 cable. */
+				ulPhyControl |= PHYCR_MDIX_FORCE;
+			}
+			/* update PHY Control Register. */
+			pxPhyObject->fnPhyWrite( xPhyAddress, phyREG_19_PHYCR, ulPhyControl );
+		}
+
+		FreeRTOS_printf( ( "+TCP: advertise: %04lX config %04lX\n", ulAdvertise, ulConfig ) );
+	}
+
+	/* Keep these values for later use. */
+	pxPhyObject->ulBCRValue = ulConfig & ~phyBMCR_ISOLATE;
+	pxPhyObject->ulACRValue = ulAdvertise;
+
+	return 0;
+}
+/*-----------------------------------------------------------*/
+
+BaseType_t xPhyFixedValue( EthernetPhy_t *pxPhyObject, uint32_t ulPhyMask )
```

Comment:
Could function comment headers please be added? This one in particular made me think of it, because I don't understand what it does based on the name.